packages/lhapdf/package.py

```diff
@@ -51,6 +51,8 @@ def patch(self):
     )
 
     depends_on("pdfsets")
+    depends_on("python")
+    depends_on("gettext")
```

Comment:
What are python and gettext needed for?

Reply:
Adding python fixed the following error for me:

```
==> Error: ProcessError: Command exited with status 1:
    '/tmp/nrowe/spack-stage/spack-stage-lhapdf-6.5.3-xq4yzkqahcszya6isvzlf23mzumbjaw2/spack-src/configure' '--prefix=/exp/sbnd/app/users/nrowe/spack_dev/spack/opt/spack/linux-almalinux9-zen3/gcc-11.4.1/lhapdf-6.5.3-xq4yzkqahcszya6isvzlf23mzumbjaw2' '--enable-low-memory' '--disable-pyext' '--disable-octave'

2 errors found in build log:
     112    checking for Python library path... -L/exp/sbnd/app/users/nrowe/spack_dev/spack/opt/spack/linux-almalinux9-x86_64_v3/gcc-11.4.1/python-3.11.9-ptsa74fzbyct53xevm6ffevlyqtyagsd/lib -lpython3.11
     113    checking for Python site-packages path... /exp/sbnd/app/users/nrowe/spack_dev/spack/var/spack/environments/testing/.spack-env/view/lib/python3.11/site-packages
     114    checking for Python platform specific site-packages path...
     115    checking python extra libraries... -lintl -ldl -lm
     116    checking python extra linking flags... -Xlinker -export-dynamic
     117    checking consistency of all components of python development environment... no
  >> 118    configure: error: in `/tmp/nrowe/spack-stage/spack-stage-lhapdf-6.5.3-xq4yzkqahcszya6isvzlf23mzumbjaw2/spack-src':
  >> 119    configure: error:
     120      Could not link test program to Python. Maybe the main Python library has been
     121      installed in some non-standard library path. If so, pass it to configure,
     122      via the LIBS environment variable.
     123      Example: ./configure LIBS="-L/usr/non-standard-path/python/lib"
     124      ============================================================================
     125       ERROR!
```
[python-spack-build-out.txt](https://github.com/user-attachments/files/16604660/python-spack-build-out.txt)

This issues seem somewhat related to [https://github.com/spack/spack/pull/37384](url). Since there isn't a python variant in this version of the package.py, it may just be assuming that variant by default since it is detecting a python already in the genie build. 

It may be easier to just update the whole lhapdf package.py to match the nominal spack version. I also noticed that lhapdf is already included in both the nominal spack and the fermilab fork [https://github.com/FNALssi/spack](url). This also seems to be true for genie itself. Am I using the preferred repos for these package.py files? Do these different versions serve different purposes?

Previously I was getting an additional error associated with not including gettext only on polaris. I am no longer getting that error if I include only the python, so I will remove that line and let you know if that becomes an issue again.

Reply:
This version of lhapdf was created before there was a version in mainline spack. If they are similar enough we could remove this one from fnal_art.

Reply:
I will try installing with the mainline spack version and make sure it works before removing the file from the PR. I will follow up on this.

Reply:
Made a discussion [here](https://github.com/FNALssi/spack/discussions/423) as I don't see an issues tab.